mutation_writer/multishard_writer.cc

```diff
@@ -116,7 +116,8 @@ future<> multishard_writer::make_shard_writer(unsigned shard) {
             consumer = _consumer,
             reader = make_foreign(std::make_unique<flat_mutation_reader_v2>(std::move(reader)))] () mutable {
         auto s = gs.get();
-        auto semaphore = std::make_unique<reader_concurrency_semaphore>(reader_concurrency_semaphore::no_limits{}, "shard_writer");
+        auto semaphore = std::make_unique<reader_concurrency_semaphore>(reader_concurrency_semaphore::no_limits{}, "shard_writer",
+                reader_concurrency_semaphore::register_metrics::no);
```

Comment:
Style nit: rather than passing bools, we could have a method register_metrics that does it and then the caller can choose to call it or not. Not worth redoing (and there's value in forcing the caller to make a choice, rather than possibly forget)./